src/cmap/connection.ts

```diff
@@ -695,7 +695,7 @@ export class CryptoConnection extends Connection {
   ): Promise<Document> {
     const { autoEncrypter } = this;
     if (!autoEncrypter) {
-      throw new MongoMissingDependencyError('No AutoEncrypter available for encryption');
+      throw new MongoAPIError('No AutoEncrypter available for encryption');
```

Comment:
This isn't really a "missing dependency" error - if anything, I think it's probably best captured by a `MongoRuntimeError` because we shouldn't be instantiating crypto connections if an autoEncrypter is not provided.  I chose to use a `MongoAPIError` because that mostly preserves the error hierarchy (`MongoAPIError` is the superclass of `MongoMissingDependencyError`).

If we decide this change is unacceptable, I can revert it and use a `ts-expect-error` instead.

Reply:
While an accurate analysis is this an acceptable API change?

> I can revert it and use a ts-expect-error here instead.

To exempt what issue exactly?

Reply:
There would be a type error because we didn't provide a cause or dependencyName to the MongoMissingDependencyError  constructor.

> While an accurate analysis is this an acceptable API change?

I think you could argue either way.  I think this is a correctness issue but I'm also happy leaving it as-is.

Reply:
It wouldn't be entirely off base to pass in `'mongodb-client-encryption'` as the dependency name here. I am leaning towards leaving the error type as is to reduce the chance of surprising anyone. I agree it may be a correctness bug but only in the errors meaning whereas fixing it has the potential to break, we can correct it in a major.

Reply:
Done.